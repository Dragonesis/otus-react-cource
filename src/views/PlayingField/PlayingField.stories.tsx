import React from 'react'
import { withReactContext } from 'storybook-react-context'
import { Story, Meta } from '@storybook/react'
import { StoreContext } from '@/services/adapters/store'
import { deckOfCardList } from '@/services/mocks'
import { PlayingField, PlayingFieldProps } from './PlayingField'

export default {
  title: 'PlayingField/PlayingField',
  component: PlayingField,
  decorators: [
    withReactContext({
      Context: StoreContext,
      initialState: {
        cardDeck: deckOfCardList[0],
        cardsInHand: [],
      },
    }),
  ],
} as Meta

export const PlayingFieldExample: Story<PlayingFieldProps> = (args) => <PlayingField {...args} />

PlayingFieldExample.args = {
  user: {
    email: '',
    name: 'User Name',
    birthday: ''
  },
}

