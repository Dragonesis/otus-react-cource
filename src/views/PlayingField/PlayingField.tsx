import React, { FC } from 'react'
import styled from '@emotion/styled'
import { User } from '@/services/models'

import { useStore } from '@/services/adapters/store'
import { Deck, Hand, Header } from './components'


export const PlayingField: FC = () => {
  const { user } = useStore()

  return (
    <>
      <Header name={user?.name || ''} />
      <Area>
        <ModDeck />
        <ModHand />
      </Area>
    </>
  )
}

const Area = styled.div`
  display: flex;
  align-items: flex-end;
  max-width: 800px;
  margin-top: auto;
  margin-left: auto;
  margin-right: auto;
`

const ModDeck = styled(Deck)`
  margin-right: 20px;
`

const ModHand = styled(Hand)`
`