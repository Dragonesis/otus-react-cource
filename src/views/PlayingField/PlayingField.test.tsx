import React from 'react'
import { render } from '@testing-library/react'

import { PlayingField } from './PlayingField'

const user = { name: 'test', email: 'wdw@dwd', birthday: '1313' }

describe('### PlayingField', () => {
  test('Renders correctly', () => {
    const { asFragment } = render(<PlayingField user={user} />)
    expect(asFragment()).toMatchSnapshot()
  })
})