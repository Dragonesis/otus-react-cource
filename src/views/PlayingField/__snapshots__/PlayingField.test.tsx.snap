// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`### PlayingField Renders correctly 1`] = `
<DocumentFragment>
  .emotion-0 {
  background-color: #3a3a3a;
  padding-top: 10px;
  padding-bottom: 10px;
}

.emotion-1 {
  max-width: 1000px;
  width: 100%;
  margin-left: auto;
  margin-right: auto;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
}

.emotion-2 {
  font-size: 18px;
  color: #fff;
  margin-right: auto;
}

.emotion-3 {
  border-radius: 6px;
  width: 100%;
  max-width: 180px;
  height: 36px;
  padding-left: 10px;
  padding-right: 10px;
  font-size: 16px;
  background-color: #fff;
  color: #3a3a3a;
  font-size: 18px;
  margin-left: 10px;
}

.emotion-3:hover,
.emotion-3:focus {
  border-color: #303030;
}

.emotion-3.size-s {
  height: 36px;
  padding-left: 10px;
  padding-right: 10px;
  font-size: 16px;
}

.emotion-3.size-m {
  height: 48px;
  padding-left: 10px;
  padding-right: 10px;
  font-size: 18px;
}

.emotion-3.size-l {
  height: 60px;
  padding-left: 16px;
  padding-right: 16px;
  font-size: 20px;
}

<div
    class="emotion-0"
  >
    <div
      class="emotion-1"
    >
      <div
        class="emotion-2"
      />
      <button
        class="emotion-3"
      >
        Новая игра
      </button>
      <button
        class="emotion-3"
      >
        Выход
      </button>
    </div>
  </div>
  .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
  max-width: 800px;
  margin-top: auto;
  margin-left: auto;
  margin-right: auto;
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

<div
    class="emotion-0"
  >
    <div
      class="emotion-1"
    />
  </div>
</DocumentFragment>
`;
